### PR TITLE
Assignment.course_id column model changes

### PR DESCRIPTION
**Pull request by marcospri**
Adds a new FK pointing to the course this assignment belongs to. 

Migration over:

- https://github.com/hypothesis/lms/pull/6519

### CODE REVIEW

File: lms/models/assignment.py [hunk 1]
```diff
@@ -72,10 +72,10 @@ class Assignment(CreatedUpdatedMixin, Base):
     title: Mapped[str | None] = mapped_column(sa.Unicode, index=True)
     """The resource link title from LTI params."""
 
-    description = sa.Column(sa.Unicode, nullable=True)
+    description: Mapped[str | None] = mapped_column(sa.Unicode)
```
Comment (marcospri) [inline lms/models/assignment.py:75]: No changes, adapting to the new typing style of SQLA 2

File: lms/models/assignment.py [hunk 2]
```diff
@@ -72,10 +72,10 @@ class Assignment(CreatedUpdatedMixin, Base):
...
     """The resource link description from LTI params."""
 
-    deep_linking_uuid: Mapped[str | None] = mapped_column(sa.Unicode, nullable=True)
+    deep_linking_uuid: Mapped[str | None] = mapped_column(sa.Unicode)
```
Comment (marcospri) [inline lms/models/assignment.py:78]: Removed the nullable info as it implied by the type definition.

**Review by robertknight — APPROVED** [review]

File: lms/models/assignment.py [hunk 3]
```diff
@@ -86,6 +86,8 @@ class Assignment(CreatedUpdatedMixin, Base):
         "AssignmentMembership", lazy="dynamic", viewonly=True
     )
 
+    course_id: Mapped[int | None] = mapped_column(sa.ForeignKey(Course.id))
```
Comment (robertknight) [inline lms/models/assignment.py:89]: This model now has both a `course_id` field and a `course` property which get results in different ways. Will the `course` property be removed in future?

Comment (marcospri) [reply lms/models/assignment.py:89]: Yes, following the steps of the PoC PR: 
https://github.com/hypothesis/lms/pull/6517/commits/c18f986effd2725a9abac99b09fe89c7396bd993#diff-71bbc86f39f14ca26fa8c345791473dcc66487d919a863426add9f9a24d6e915L95

That would be the last step of this train of PRs :+1:

**Pull request merged**